commons/src/main/java/io/hyscale/commons/commands/CommandExecutor.java

```diff
@@ -173,32 +177,44 @@ private static CommandResult _execute(String command, File file, String dir, Str
             throw e;
         }
         if (readOutput) {
+            while(!copyOutput.isDone) {
+                // wait till copy is over
+            }
             cmdResult.setCommandOutput(strWriter.toString());
         }
         cmdResult.setExitCode(exitCode);
         return cmdResult;
     }
-
     /**
-     * Copy output from process input stream to string writer
-     *
-     * @param process
-     * @param strWriter
-     * @return is thread started
+     *  Copy output from process input stream to string writer
+     *  Wait till copying is done
      */
-    private static boolean copyOutput(Process process, StringWriter strWriter) {
-        return ThreadPoolUtil.getInstance().execute(() -> {
-            try {
-                do{
-                    IOUtils.copy(process.getInputStream(), strWriter, StandardCharsets.UTF_8);
-                } while (process.isAlive());
-            } catch (IOException e) {
-                logger.error("Error while reading command output", e);
-            }
-        });
+    static class CopyOutput{
+        
+        private boolean isDone = false;
+        
+        /**
+         * Copy output from process input stream to string writer
+         *
+         * @param process
+         * @param strWriter
+         * @return is thread started
+         */
+        public boolean performOperation(Process process, StringWriter strWriter) {
+            return ThreadPoolUtil.getInstance().execute(() -> {
+                try {
+                    do{
+                        IOUtils.copy(process.getInputStream(), strWriter, StandardCharsets.UTF_8);
+                    } while (process.isAlive());
+                } catch (IOException e) {
+                    logger.error("Error while reading command output", e);
+                } finally {
+                    isDone = true;
+                }
```

Comment:
Need to have some step wait so that when process is not alive and we are not done copying the inputstream